python/tskit/tables.py

```diff
@@ -2470,3 +2470,172 @@ def drop_index(self):
         currently indexed this method has no effect.
         """
         self.ll_tables.drop_index()
+
+    def add_time(self, dt):
+        """
+        This function modifies a table collection IN PLACE by adding ``dt`` to
+        all times in the node table.
+        """
+        nodes_dict = self.nodes.asdict()
+        nodes_dict['time'] = nodes_dict['time'] + dt
+        self.nodes.set_columns(**nodes_dict)
+        migrations_dict = self.migrations.asdict()
+        migrations_dict['time'] = migrations_dict['time'] + dt
+        self.migrations.set_columns(**migrations_dict)
+
+    def graft(self, other, node_map, check_overlap=False):
+        """
+        Returns tables by grafting the table collection ``other`` onto this
+        table collection along the nodes specified in the dictionary ``node_map``,
+        which should be a dictionary whose keys are node IDs in ``other`` and values
+        are the equivalent node IDs in this set of tables.
+        Any populations of nodes that are newly added to this table collection
+        are considered new in the grafted tables, even if their IDs are the same.
+        Map is returned (TODO: say exactly what).
+        If ``check_overlap`` is True, then check whether the two tables, simplified
```

Comment:
In practise this check will be quicker than the rest of the method, since we're looping over full tables of edges and nodes in Python. Unless this ends up being implemented in C, running simplify will make no real difference, I think. (So, perhaps we should just do it, and add the option to not do it later if it becomes a bottleneck?)